File: Day-01-Arrays/JavaScript/217_Contains_Duplicate.js
/*
Problem: 217. Contains Duplicate

Pattern: Hash Set

Time Complexity: O(n)
Space Complexity: O(n)
*/

var containsDuplicate = function(nums) {

    const seen = new Set();

    for (const num of nums) {

        if (seen.has(num))
            return true;

        seen.add(num);
    }

    return false;
};

File: Day-01-Arrays/Python/217_Contains_Duplicate.py
"""
Problem: 217. Contains Duplicate

Pattern: Hash Set

Time Complexity: O(n)
Space Complexity: O(n)
"""

class Solution:
    def containsDuplicate(self, nums: list[int]) -> bool:

        seen = set()

        for num in nums:

            if num in seen:
                return True

            seen.add(num)

        return False